test/api/users/test-user-manager.js

```diff
@@ -0,0 +1,167 @@
+const mockgoose = require('mockgoose');
+const mongoose = require('mongoose');
+const chai = require('chai');
+const userManager = require('../../../api/users/user-manager');
+const userModels = require('../../../api/users/user-models');
+const testUsers = require('../../core/users');
+
+const expect = chai.expect;
+const should = chai.should();
+const Admin = userModels.Admin;
+const Student = userModels.Student;
+
+chai.use(require('chai-shallow-deep-equal'));
+chai.use(require('chai-moment'));
+
+mockgoose(mongoose);
+
+describe('userManager', () => {
+  // Connect to the database.
+  before((done) => { mongoose.connect('', done); });
+
+  // Clear the database before each test case.
+  beforeEach((done) => {
+    mockgoose.reset();
+    done();
+  });
+
+  after((done) => { mongoose.unmock(done); });
+
+  describe('#createUser', () => {
+    describe('role: student', () => {
+      it('should pass a uid to the callback', (done) => {
+        const student = testUsers.student000;
+
+        userManager.createUser(student, (createErr, uid) => {
+          expect(createErr).to.be.null;
+          expect(uid).to.be.a('string');
+
+          Student.findById(uid, (studentErr, foundStudent) => {
+            expect(studentErr).to.be.null;
+            foundStudent.comparePassword(student.password, (passwordErr, isMatch) => {
+              expect(isMatch).to.be.true;
+            });
+            student.password = foundStudent.password; // foundStudent has a hashed password
+            expect(foundStudent).to.shallowDeepEqual(student);
+            done();
+          });
+        });
+      });
+
+      it('should pass an error to the callback', (done) => {
+        const student = {
+          email: 'test@test.com',
+          nameame: 'Test',
```

Comment:
Didn't catch the `nameame` actually. The `studentId` is the error to catch. This test actually needs to be rewritten to use the `core/users` file.